hbase-server/src/main/java/org/apache/hadoop/hbase/master/AssignmentManager.java

```diff
@@ -1995,97 +1995,100 @@ private void unassign(final HRegionInfo region,
+              if (t instanceof FailedServerException) {
+                sleepTime = 1L + conf.getInt(RpcClient.FAILED_SERVER_EXPIRY_KEY,
                   RpcClient.FAILED_SERVER_EXPIRY_DEFAULT);
-          } else {
-            // RS is already processing this region, only need to update the timestamp
-            LOG.debug("update " + state + " the timestamp.");
-            state.updateTimestampToNow();
-            if (maxWaitTime < 0) {
-              maxWaitTime =
-                  EnvironmentEdgeManager.currentTime()
-                      + conf.getLong(ALREADY_IN_TRANSITION_WAITTIME,
-                        DEFAULT_ALREADY_IN_TRANSITION_WAITTIME);
-            }
-            long now = EnvironmentEdgeManager.currentTime();
-            if (now < maxWaitTime) {
-              LOG.debug("Region is already in transition; "
-                + "waiting up to " + (maxWaitTime - now) + "ms", t);
-              sleepTime = 100;
-              i--; // reset the try count
-              logRetries = false;
+              } else {
+                // RS is already processing this region, only need to update the timestamp
+                LOG.debug("update " + state + " the timestamp.");
+                state.updateTimestampToNow();
+                if (maxWaitTime < 0) {
+                  maxWaitTime = EnvironmentEdgeManager.currentTime() + conf.getLong(
+                    ALREADY_IN_TRANSITION_WAITTIME, DEFAULT_ALREADY_IN_TRANSITION_WAITTIME);
+                }
+                long now = EnvironmentEdgeManager.currentTime();
+                if (now < maxWaitTime) {
+                  LOG.debug("Region is already in transition; " + "waiting up to "
+                      + (maxWaitTime - now) + "ms",
+                    t);
+                  sleepTime = 100;
+                  i--; // reset the try count
+                  logRetries = false;
+                }
+              }
             }
-          }
-        }
-
-        try {
-          if (sleepTime > 0) {
-            Thread.sleep(sleepTime);
-          }
-        } catch (InterruptedException ie) {
-          LOG.warn("Failed to unassign "
-            + region.getRegionNameAsString() + " since interrupted", ie);
-          Thread.currentThread().interrupt();
-          if (state != null) {
-            regionStates.updateRegionState(region, State.FAILED_CLOSE);
-          }
-          return;
-        }
 
         if (logRetries) {
-          LOG.info("Server " + server + " returned " + t + " for "
-            + region.getRegionNameAsString() + ", try=" + i
-            + " of " + this.maximumAttempts, t);
+          LOG.info("Server " + server + " returned " + t + " for " + region.getRegionNameAsString()
+              + ", try=" + i + " of " + this.maximumAttempts,
+            t);
           // Presume retry or server will expire.
         }
       }
+      // If sleepTime is not set by any of the cases, set it to sleep for
```

Comment:
nit: Can you please add a few comments on the code flow so that its easier for other to follow?